### ISSUE TITLE
Difficulty closing palette search area for screen reader users

### ISSUE THREAD

**Issue by jmwmusic** [musescore/MuseScore#33896]
### Issue type

Accessibility issue (e.g. for keyboard-only or screen reader users)

### Description with steps to reproduce

Palette search doesn’t close when an item is selected or clear is clicked.  Possibly this is a pop-up that stays open by design, but how to close it is very unclear to a screen reader user with only keyboard navigation.
it can be dismissed with Esc, but this is not obvious.  could we have speech to say what it is and how to close it, or a close button.  

STEPS:
1.	 Navigate to the Palette search button and press enter.
2.	Type for something to search for e.g. “ornament” and press enter.
3.	Press Tab to hear “ornaments palette”, expand this, arrow down to turn and press enter.
4.	Notice as expected focus jumps to the note on the score where the turn has been added.
5.	Press Shift + Tab twice to get back to the palette area and notice that the search is still open and it is not obvious how to close this and return to the usual palette area.
6.	Press Shift Tab again and change to properties panel.
7.	Select Palettes again, and the search area still seems to be open.
8.	Press Esc and it closes and returns to usual.

I’m not sure how with a mouse others close this, but it does not seem at all obvious to a screen reader how to get back to the original Palette area.

### Supporting files, videos and screenshots

N/A

### In which versions of MuseScore Studio is this issue present?

4.7.3

### Regression

I was unable to check

### Operating system

Windows 11 with NVDA

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible
